### ISSUE TITLE
[Bug]  shuffle server have blocked threads

### ISSUE THREAD

**Issue by smlHao** [apache/incubator-uniffle#1026]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-uniffle/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

 hi， when huge table join huge table， shuffle server have blocked threads ， Is it right?

server conf :  
rss.rpc.server.port 20000
rss.jetty.http.port 20001
rss.storage.basePath /app/rss-0.7.1/data
rss.storage.type MEMORY_LOCALFILE_HDFS
rss.coordinator.quorum 172.100.3.70:19999,172.100.3.71:19999,172.100.3.72:19999
rss.server.disk.capacity 50g

rss.server.flush.thread.alive 30
rss.server.flush.threadPool.size 10
rss.server.buffer.capacity 40g
rss.server.read.buffer.capacity 20g
rss.server.heartbeat.interval 10000
rss.rpc.message.max.size 1073741824
rss.server.preAllocation.expired 120000
rss.server.commit.timeout 600000
rss.server.app.expired.withoutHeartbeat 120000
rss.server.flush.cold.storage.threshold.size 512m

rss client conf :

spark.shuffle.manager=org.apache.spark.shuffle.RssShuffleManager
spark.rss.coordinator.quorum=172.100.3.70:19999,172.100.3.71:19999,172.100.3.72:19999
# in production need to chagne to MEMORY_LOCALFILE_HDFS
spark.rss.storage.type=MEMORY_LOCALFILE_HDFS
spark.rss.remote.storage.path=hdfs://ns1/rss/sml

![image](https://github.com/apache/incubator-uniffle/assets/3837738/4fdb9724-4e6d-497a-96dc-01ee45ec9f75)

![image](https://github.com/apache/incubator-uniffle/assets/3837738/97f9a59b-5909-4db6-aef0-9ac89805ab29)

![image](https://github.com/apache/incubator-uniffle/assets/3837738/3ed319e3-f1d9-417c-a570-b587ab70850e)

the executor have no daemon thread holding and hava no error log

![image](https://github.com/apache/incubator-uniffle/assets/3837738/1baad065-8769-4092-82b6-5dbb24478167)

![image](https://github.com/apache/incubator-uniffle/assets/3837738/76b81e9d-1729-40f0-9bec-2c0da7052a79)

### Affects Version(s)

0.7.1

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

**Comment by zuston** [comment]
What do you want to report? The app failed?

**Comment by smlHao** [comment]
> What do you want to report
     1 . when huge table join huge table， shuffle server have blocked threads ， Is it right?  
     2 . the executor have no daemon thread and  holding ,  seem to hold on senddata to uniffle server , is my conf need to adjust ?
> The app failed ?
    after running 2 hours, seem no failed but executors holding , executor logs not update ,driver log only have uniffle heatbeat log


Do you have  performance tuning  about spark sql  huge table join ?

**Comment by zuston** [comment]
No. But I have tuning with the huge partition. Firstly, we should find out the root cause. Please tell what happened for you app.

**Comment by smlHao** [comment]
> No. But I have tuning with the huge partition. Firstly, we should find out the root cause. Please tell what happened for you app.

@zuston  thanks !!!  yes, you are right ,  I found that : 

executor log  long time no update :
![image](https://github.com/apache/incubator-uniffle/assets/3837738/6d40a4be-77b4-4365-8e16-8f00359308cd)

then I check the executor  stack ,  find there are no daemon threads  WAITTING, seem to holding on senddata: 

https://user-images.githubusercontent.com/3837738/254862640-1baad065-8769-4092-82b6-5dbb24478167.png

then I analysis the shuffle server  stack , find there are  threads  BLOCKED

 myapp process seem  no change,  But I can`t find out the root cause , Do you have some steps to help me ?


>  tuning with the huge partition 
    Do you have documents help me do this ?

**Comment by zuston** [comment]
Can you check the shuffle-server and executor GC? Why not using the spark ui? And I think if you want to analysis, it's better to show the metrics into dashboard.

**Comment by smlHao** [comment]
> Can you check the shuffle-server and executor GC? Why not using the spark ui? And I think if you want to analysis, it's better to show the metrics into dashboard.
1 . check the shuffle-server and executor GC ： 
    shuffle-server no full gc , but  jvm_memory_bytes_used is  close to XMX_SIZE="60g" ：
![image](https://github.com/apache/incubator-uniffle/assets/3837738/5f717ed0-f8c7-45f1-b6eb-1e23ab2f95d6)
 executor gc seem normal ：
![image](https://github.com/apache/incubator-uniffle/assets/3837738/afe39443-6f8f-48e0-a6c9-16af2e29fa90)

2.  using the spark ui ： dont`t exported the port  and need use vpn
3.  metrics into dashboard : had show some metrics , feel shuffle server had too many blocked thread not normal ，but can`t find why 
    
![image](https://github.com/apache/incubator-uniffle/assets/3837738/f6099093-7e11-4169-8924-e36556aad997)
![image](https://github.com/apache/incubator-uniffle/assets/3837738/1ba65eb2-a6d7-40de-8aea-c9c28b4c4d48)
![image](https://github.com/apache/incubator-uniffle/assets/3837738/2ad58587-6c40-4e68-b8a3-d88b5bc021db)

@zuston  How do you  tuning with the huge partition ？  Could you help me do this ?
